// @edisims/database/src/permissions.ts
import { PermissionKey } from "@prisma/client";

export const OWNER_ROLE = "OWNER";
export const CASHIER_ROLE = "CASHIER";
export const STORE_STAFF_ROLE = "STORE_STAFF";
export const ONLINE_MANAGER_ROLE = "ONLINE_MANAGER";

export const BUILTIN_ROLES = [
  OWNER_ROLE,
  CASHIER_ROLE,
  STORE_STAFF_ROLE,
  ONLINE_MANAGER_ROLE,
] as const;

export const ALL_PERMISSIONS = Object.values(PermissionKey);

export const PERMISSION_LABELS: Record<PermissionKey, string> = {
  [PermissionKey.PRODUCTS_VIEW]: "View products",
  [PermissionKey.PRODUCTS_CREATE]: "Create products",
  [PermissionKey.PRODUCTS_EDIT]: "Edit products",
  [PermissionKey.PRODUCTS_DEACTIVATE]: "Deactivate products",
  [PermissionKey.PRODUCTS_SELL]: "Sell products",
  [PermissionKey.PRODUCTS_RESERVE]: "Reserve stock",
  [PermissionKey.PRODUCTS_RESTOCK]: "Restock products",
  [PermissionKey.PRODUCTS_EXPORT]: "Export product list",
  [PermissionKey.CATEGORIES_MANAGE]: "Manage categories",
  [PermissionKey.PRODUCT_TYPES_MANAGE]: "Manage product types",
  [PermissionKey.STOCK_RECEIVE]: "Receive stock",
  [PermissionKey.STOCK_HISTORY_VIEW]: "View stock history",
  [PermissionKey.SALES_CREATE]: "Record sales",
  [PermissionKey.SALES_VIEW]: "View sales",
  [PermissionKey.SALES_EXPORT]: "Export sales",
  [PermissionKey.RESERVATIONS_MANAGE]: "Manage reservations",
  [PermissionKey.USERS_MANAGE]: "Manage users",
  [PermissionKey.ROLES_MANAGE]: "Manage role permissions",
  [PermissionKey.SETTINGS_EMAILS]: "Manage alert emails",
  [PermissionKey.REPORTING_VIEW]: "View reports",
};

const staffPermissions: PermissionKey[] = [
  PermissionKey.PRODUCTS_VIEW,
  PermissionKey.PRODUCTS_CREATE,
  PermissionKey.PRODUCTS_EDIT,
  PermissionKey.PRODUCTS_SELL,
  PermissionKey.PRODUCTS_RESERVE,
  PermissionKey.PRODUCTS_RESTOCK,
  PermissionKey.PRODUCTS_EXPORT,
  PermissionKey.CATEGORIES_MANAGE,
  PermissionKey.PRODUCT_TYPES_MANAGE,
  PermissionKey.STOCK_RECEIVE,
  PermissionKey.STOCK_HISTORY_VIEW,
  PermissionKey.SALES_CREATE,
  PermissionKey.SALES_VIEW,
  PermissionKey.SALES_EXPORT,
  PermissionKey.RESERVATIONS_MANAGE,
  PermissionKey.REPORTING_VIEW,
];

const cashierPermissions: PermissionKey[] = [
  PermissionKey.PRODUCTS_VIEW,
  PermissionKey.PRODUCTS_EDIT,
  PermissionKey.PRODUCTS_SELL,
  PermissionKey.PRODUCTS_RESERVE,
  PermissionKey.PRODUCTS_EXPORT,
  PermissionKey.SALES_CREATE,
  PermissionKey.SALES_VIEW,
  PermissionKey.SALES_EXPORT,
  PermissionKey.RESERVATIONS_MANAGE,
  PermissionKey.REPORTING_VIEW,
];

export const DEFAULT_ROLE_PERMISSIONS: Record<string, PermissionKey[]> = {
  [OWNER_ROLE]: ALL_PERMISSIONS,
  [STORE_STAFF_ROLE]: staffPermissions,
  [CASHIER_ROLE]: cashierPermissions,
  [ONLINE_MANAGER_ROLE]: [PermissionKey.PRODUCTS_VIEW],
};

export const isOwnerRole = (role: string) => role === OWNER_ROLE;
